### PR TITLE
rofi: add options to for location, xoffset & yoffset

### PR DESCRIPTION
**Pull request by mogria**
Just added the options `location`, `xoffset` and `yoffset` to be able to position rofi at the desired location

### CODE REVIEW

File: modules/programs/rofi.nix [hunk 1]
```diff
@@ -195,6 +195,24 @@ in
       description = "Whether to run rofi fullscreen.";
     };
 
+    location = mkOption {
+      default = null;
+      type = types.nullOr types.int;
+      description = "Location on screen (1-8)";
```
Comment (rycee) [inline modules/programs/rofi.nix]: Could have `type = types.nullOr (types.enum [1 2 3 4 5 6 7 8]);` here.

Comment (rycee) [reply modules/programs/rofi.nix]: Hmm, actually I think it in this case is fair to simply set the default to 0 and include the pattern in the description. Something like

```nix
    location = mkOption {
      default = 0;
      type = types.enum (range 0 8);
      description = ''
        Location on screen following the pattern

        <informaltable frame="none"><tgroup cols="1"><tbody>
        <row><entry>1</entry><entry>2</entry><entry>3</entry></row>
        <row><entry>8</entry><entry>0</entry><entry>4</entry></row>
        <row><entry>7</entry><entry>6</entry><entry>5</entry></row>
        </tbody></tgroup></informaltable>
      '';
    };
```

File: modules/programs/rofi.nix [hunk 2]
```diff
@@ -195,6 +195,24 @@ in
...
+    xoffset = mkOption {
+      default = null;
+      type = types.nullOr types.int;
+      description = "Offset in x-axis from chosen location";
+    };
+
+    yoffset = mkOption {
+      default = null;
+      type = types.nullOr types.int;
+      description = "Offset in y-asxis from chosen location";
```
Comment (rycee) [inline modules/programs/rofi.nix]: s/y-asxis/y-axis/

**Comment by rycee** [comment]
Thanks for the contribution! I've made a few comments. Once fixed I'd be happy to merge.

**Comment by mogria** [comment]
I've added another commit.
Thank you for your comments, helps me get better using nix :)

**Comment by uvNikita** [comment]
Maybe it's even better to have something like `types.enum [ "top-left" "top", "top-right"... ];` instead of numbers? Makes it a bit more human readable :)

**Comment by rycee** [comment]
Yeah, I was thinking the same but wasn't certain. I certainly think it would be more intuitive :-)

@mogria If you'd like to try out the fancy naming scheme then you can practice your Nix skills by adding an attribute set in the `let` section at the top of the file:

```nix
locationsMap = {
  center = 0;
  "top left" = 1;
  top = 2;
  "top right" = 3;
  # …
};
```

and you can define the type using `builtins.attrNames`.

**Comment by mogria** [comment]
@rycee Thanks for the advice! I've updated it now.

Should I rebase, squash the commits and force push, before merging?

File: modules/programs/rofi.nix [hunk 3]
```diff
@@ -195,6 +206,24 @@ in
...
+    xoffset = mkOption {
+      default = 0;
+      type = types.int;
+      description = "Offset in the x-axis in pixels relative to the chosen location";
+    };
+
+    yoffset = mkOption {
+      default = 0;
+      type = types.int;
+      description = "Offset in y-axis in pixels relative to the chosen location";
```
Comment (rycee) [inline modules/programs/rofi.nix]: s/Offset in y-axis/Offset in the y-axis/

**Comment by rycee** [comment]
@mogria Yes, please squash and use `rofi: add options for location, xoffset & yoffset` as commit message.

**Comment by rycee** [comment]
I had some spare time so I squashed and rebased this into master 6833f96c147779fe197657bbcc4bd323324cd6c3. Many thanks for the improvements!

**Pull request closed without merging**